real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/main/RobobuggyConfigFile.java

```diff
@@ -30,20 +44,267 @@
+		}
+		//TODO add other settings as they are created 
+		catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Helper function which removes the first and last elements from a string
+	 * @param input
+	 * @return quote free string
+	 */
+	private static String removeQuotes(String input){
+		//the string is too short
+		if(input.length() < 2){
+			return "";
+		}
+		return input.substring(1, input.length() -1);
+	}
+	
+	/**
+	 * Attempts to load values for configuration settings based on the configuration file
+	 * If the configuration file cannot be read then the old values will be used 
+	 */
+	public static void loadConfigFile(){
+		try {
+			JsonObject configJson = Util.readJSONFile(getCONFIG_FILE());
+			setCOM_PORT_IMU(removeQuotes(configJson.get("COM_PORT_IMU").toString()));
+			System.out.println("comPORT:"+getCOM_PORT_IMU());
+			setCOM_PORT_GPS(removeQuotes(configJson.get("COM_PORT_GPS").toString()));
+			setCOM_PORT_RBSM(removeQuotes(configJson.get("COM_PORT_RBSM").toString()));
+			setPORT_VISION(removeQuotes(configJson.get("PORT_VISION").toString()));
+			setWAYPOINT_SOURCE_LOG_FILE(removeQuotes(configJson.get("WAYPOINT_SOURCE_LOG_FILE").toString()));
+			setPLAY_BACK_SOURCE_FILE(removeQuotes(configJson.get("PLAY_BACK_SOURCE_FILE").toString()));
+
+			
+		} catch (UnsupportedEncodingException | FileNotFoundException e) {
+			new RobobuggyLogicNotification("could not load Part of configFile"+e.toString(), RobobuggyMessageLevel.WARNING);
+		}
+	}
+
+	/**
+	 * evaluates to the current value of imuEnabled
+	 * @return
+	 */
+	public static boolean isImuEnabled() {
+		return imuEnabled;
+	}
+
+	/**
+	 * sets imuEnabled to a new value
+	 * @param imuEnabled
+	 */
+	public static void setImuEnabled(boolean imuEnabled) {
+		RobobuggyConfigFile.imuEnabled = imuEnabled;
+	}
+
+	/**
+	 * evaluates to the current value of dataPlayBack
+	 * @return
+	 */
+	public static boolean isDataPlayBack() {
+		return DATA_PLAY_BACK;
+	}
+
+	/**
+	 * evaluates to the current value of comPortImu 
+	 * @return
+	 */
+	public static String getCOM_PORT_IMU() {
+		return comPortImu;
+	}
+
+	/**
+	 * sets a new value to comPortImu
+	 * @param cOM_PORT_IMU
+	 */
+	public static void setCOM_PORT_IMU(String cOM_PORT_IMU) {
+		comPortImu = cOM_PORT_IMU;
+	}
+
+	/**
+	 * evaluates to the current value of configFile
+	 * @return
+	 */
+	public static String getCONFIG_FILE() {
+		return configFile;
+	}
+
+	/**
+	 * sets a new value to configFile
+	 * @param cONFIG_FILE
+	 */
+	public static void setCONFIG_FILE(String cONFIG_FILE) {
+		configFile = cONFIG_FILE;
+	}
+
+	/**
+	 * evaluates to the current value of waypointSourceLogFile
+	 * @return
+	 */
+	public static String getWAYPOINT_SOURCE_LOG_FILE() {
+		return waypointSourceLogFile;
+	}
+
+	/**
+	 * sets a new value to waypointSourceLogFile
+	 * @param wAYPOINT_SOURCE_LOG_FILE
+	 */
+	public static void setWAYPOINT_SOURCE_LOG_FILE(
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method names should not contain underscores](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2254452545/issues/source?bid=2938665&fileBranchId=3180106#l192)